### ISSUE TITLE
[BUG] Failures with Java security.policy when using repository-azure plugin managed identity

### ISSUE THREAD

**Issue by c3-davidtran** [opensearch-project/OpenSearch#15020]
### Describe the bug

I've noticed 2 bugs related to java security.policy : 

**Managed Identity authentication** :
SInce Opensearch 2.15, we are now able to authenticate with Azure Managed identity when using repository-azure plugin.
As recommended by Azure, the privileged approach is to use workload identity with a federated Identity.
https://azure.github.io/azure-workload-identity/docs/introduction.html

This will generate a federated token when the pod starts, allowing authentication.
The token is located at path : /var/run/secrets/azure/tokens/azure-identity-token
So when enabling authentication using Managed identity, the plugin fails because it cannot read this file.
To solve it, we need to add a read permission on the java security.policy.
e.g 
```
permission java.io.FilePermission "/var/run/secrets/azure/tokens/azure-identity-token", "read";
```

### Related component

Plugins

### To Reproduce

Deploy Azure wokload identity controller
Deploy an opensearch cluster :
- Make sure to use in opensearch.yaml : **azure.client.default.token_credential_type: "managed_identity"**
- Make sure pods authenticate using federated identity

### Expected behavior

Pods java process read /var/run/secrets/azure/tokens/azure-identity-token

### Additional Details

**Plugins**
Please list all plugins currently enabled.

**Screenshots**

**Host/Environment (please complete the following information):**

**Additional context**

**Comment by cwperks** [comment]
@c3-davidtran [repository-azure](https://github.com/opensearch-project/OpenSearch/tree/main/plugins/repository-azure) is an OpenSearch plugin which has its own [plugin-security.policy](https://github.com/opensearch-project/OpenSearch/blob/main/plugins/repository-azure/src/main/plugin-metadata/plugin-security.policy) file. This would be the place to add an entry to grant permission to read the file for the plugin.

Would it be possible to test an added entry to that file? After modifying the `plugin-security.policy` you can create a local snapshot using `./gradlew assemble` from the root directory of the plugin. The output of `./gradlew assemble` would be a zip file in the `build/distributions/` directory.

**Comment by peternied** [comment]
[Triage - attendees [1](https://github.com/peternied) [2](https://github.com/reta) [3](https://github.com/andrross) [4](https://github.com/enjoychange) [5](https://github.com/lukas-vlcek/)]
@c3-davidtran Thanks for creating this issue, what do you think about creating a pull request to fix this?

**Comment by cthtrifork** [comment]
@reta Can you expand upon your thoughts on solving it through a setting?

@chengwushi-netapp Did you not hit this issue while implementing https://github.com/opensearch-project/OpenSearch/pull/12559 ?

**Comment by reta** [comment]
> @reta Can you expand upon your thoughts on solving it through a setting?

Absolutely, thanks @cthtrifork . The idea is to introduce the setting like this:

```
    /** The identity token file for connecting to Azure. */
    static final Setting.AffixSetting<String> IDENTITY_TOKEN_FILE_SETTING = Setting.affixKeySetting(
        PREFIX,
        "identity_token_file",
        key -> SecureSetting.simpleString(key, Property.NodeScope)
    );
```

The lookup in default location (`/var/run/secrets/azure/tokens/azure-identity-token`) should be replaced by this setting (the corresponding Azure SDK configuration setting / system property is AZURE_FEDERATED_TOKEN_FILE). The `/var/run/secrets/azure/tokens/azure-identity-token` could be soft linked to `${OPENSEARCH_PATH_CONFIG}/azure-identity-token` and the new setting could point to the link (the location is relative to ${OPENSEARCH_PATH_CONFIG}):

```
azure.default.identity_token_file: azure-identity-token
```

The plugin should be able to read the `azure-identity-token` without any security violations.

**Comment by chengwushi-netapp** [comment]
>@chengwushi-netapp Did you not hit this issue while implementing https://github.com/opensearch-project/OpenSearch/pull/12559 ?

Hello @cthtrifork, thank you for the question. When I was working on the implementation, I did not encounter the issue you're referring to. This is because the scope of the original ticket, #12423, did not include Workload Identity. As such, I didn't delve into the workings of Workload Identity and might have overlooked that it could be set up with Managed Identity. Consequently, I never tested the setup of Workload Identity with the Managed Identity implemented in #12559.

**Comment by cthtrifork** [comment]
@chengwushi-netapp Thank you for clarifying. The landscape in azure for managed identities is quite confusing, as it has had many names and paradigms! Now i understand why you have not had the issue.

So are there any other approaches than creating a custom Dockerfile with a `run ln -s #softlink` instruction + adding the new setting? It seems like a hack.

It almost seems like updating https://github.com/opensearch-project/OpenSearch/blob/main/plugins/repository-azure/src/main/plugin-metadata/plugin-security.policy is a better approach.

What are people thoughts

**Comment by reta** [comment]
> What are people thoughts? @chengwushi-netapp @reta\n@peternied any opinions on approaches?

Thanks @cthtrifork , patching the security policy is not recommended (it opens up yet another can of worms), soft link is the safest option I believe and is aligned with other plugins that support managed identity

**Comment by c3-davidtran** [comment]
Hello @cwperks @peternied @reta  
It's been quite some time since we last discussed this issue. We are still encountering this issue in version 2.18 so would you 
be aware of any updates ? (e.g solution has been decided, implementation is in progress etc... )

**Comment by reta** [comment]
> Hello @cwperks @peternied @reta  
> It's been quite some time since we last discussed this issue. We are still encountering this issue in version 2.18 so would you 
> be aware of any updates ? (e.g solution has been decided, implementation is in progress etc... )

@c3-davidtran the implementation path seems to be clear but afaik, no one is working on that, if you are willing to submit a pull request,  that would be much appreciated,  thank you!

**Comment by tillwulfram** [comment]
@c3-davidtran @cwperks Do you have any workaround in place for that? We would like to deploy in azure and access_keys are blocked by design so we need to use the msi but can't because of this bug.. Any recommendations oder workarounds currently?
